content/api/hub/architecture_cqrs_go.md

```diff
@@ -0,0 +1,27 @@
+title = "CQRS with Go"
+template = "render_hub_content_body"
+date = "2024-04-03T13:50:00Z"
+content-type = "text/plain"
+tags = ["http", "go", "architecture"]
+
+[extra]
+author = "Thorsten Hans"
+type = "hub_document"
+category = "Architecture"
+language = "Go"
+created_at = "2024-04-03T13:50:00Z"
+last_updated = "2024-04-03T13:50:00Z"
+spin_version = ">=v2.4.0"
+summary = "A Command and Query Responsibility Segregation (CQRS) implementation in Go"
+url = "https://github.com/fermyon/enterprise-architectures-and-patterns/tree/main/cqrs-go"
+keywords = "Architecture, HTTP, CQRS"
+
+---
+
+This is a simple Command and Query Responsibility Segregation (CQRS) implementation written in Go.
+
+### What is Command and Query Responsibility Segregation (CQRS)
+
+CQRS is a software architectural pattern that separates the responsibility of handling commands (requests that change the system's state) from handling queries (requests that fetch data without modifying state). In a CQRS architecture, commands are handled by a separate component or layer known as the Command side, while queries are handled by another component or layer called the Query side. This segregation allows each side to be optimized independently, as they often have different scalability, performance, and optimization requirements.
+
+On the Command side, operations are focused on enforcing business rules, validation, and updating the state of the system. This side typically utilizes a domain-driven design approach to model the business logic effectively. On the Query side, the emphasis is on efficiently retrieving data to fulfill read requests from clients. This side often employs de-normalized data models and specialized data storage mechanisms optimized for fast read access. By separating the concerns of commands and queries, CQRS promotes a clearer separation of concerns and can lead to improved scalability, performance, and maintainability in complex software systems.
```

Comment:
Do we want to add a small CTA section at the bottom, such as . . . 

```suggestion
On the Command side, operations are focused on enforcing business rules, validation, and updating the state of the system. This side typically utilizes a domain-driven design approach to model the business logic effectively. On the Query side, the emphasis is on efficiently retrieving data to fulfill read requests from clients. This side often employs de-normalized data models and specialized data storage mechanisms optimized for fast read access. By separating the concerns of commands and queries, CQRS promotes a clearer separation of concerns and can lead to improved scalability, performance, and maintainability in complex software systems.

To get started, check out our simple [CQRS implementation](https://github.com/fermyon/enterprise-architectures-and-patterns/tree/main/cqrs-go) with Spin apps on GitHub
```

Reply:
Don't know. There are already two links to the sample in the Hub 

<img width="1174" alt="image" src="https://github.com/fermyon/developer/assets/357884/8e0aadc7-916f-4373-aa57-f2216671ddb4">